mirgecom/initializers.py

```diff
@@ -262,6 +263,100 @@ def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
                               momentum=mom)
 
 
+class DoubleMachReflection:
+    r"""Implement the double shock reflection problem.
+
+        - Woodward and Collela
+
+    The inital condition is defined
+
+    .. math::
+
+        (\rho,u,v,P) =
+
+    This function only serves as an initial condition
+
+    .. automethod:: __init__
+    .. automethod:: __call__
+    """
+
+    def __init__(
+            self, dim=2, shock_location=1.0/6.0, shock_speed=4.0
+    ):
+        """Initialize initial condition options.
+
+        Parameters
+        ----------
+        dim: int
+           dimension of domain, must be 2
+        shock_location: float
+           location of shock
+        shock_speed: float
+           shock speed, Mach number
+        """
+        self._shock_location = shock_location
+        self._dim = dim
+        self._shock_speed = shock_speed
+
+    def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
+        """
+        Create the 1D Sod's shock solution at locations *x_vec*.
+
+        Parameters
+        ----------
+        t: float
+            Current time at which the solution is desired (unused)
+        x_vec: numpy.ndarray
+            Nodal coordinates
+        eos: :class:`mirgecom.eos.GasEOS`
+            Equation of state class to be used in construction of soln (if needed)
+        """
+        assert self._dim == 2, "only defined for dim=2"
```

Comment:
Exception instead of assert (if dim is even needed).

Reply:
dim is really only used currently for this check so it doesn't crash when ran with 1 or 3D. Don't know if we want to assume the user knows that the case is setup to be 2D?

Reply:
I think you could just remove the `dim` argument from `__init__()` (as in `Vortex2D`), and then document that it's 2D-only. @MTCam agree?

Reply:
I think it'd still be good to toss an error when used in other than 2d.  Like:

```python
# Fail if thing numdim other than 2 
if(len(x_vec)) != 2:
    raise ValueError("Blah blah error blah")
```

Reply:
Okay, I am working on some revisions to this case initialization I can add that. dim is also used by the join_conserved though, but that could be hardcoded to 2 as well?

Reply:
Removed dim argument here: https://github.com/illinois-ceesd/mirgecom/pull/213/commits/974bcd47752630f73c6b9a536352b4bc128a5b2c